### PR TITLE
PYTHON-2234: When mongocryptd spawn fails, the driver does not indicate what it tried to spawn

### PR DESCRIPTION
**Pull request by guanlinzhou**
Catching `FileNotFound` error in spawning `Mongocryptd`, and provide a more useful message

### CODE REVIEW

File: pymongo/daemon.py [hunk 1]
```diff
@@ -54,12 +54,16 @@ def _silence_resource_warning(popen):
...
+        try:
+            with open(os.devnull, 'r+b') as devnull:
+                popen = subprocess.Popen(
+                    args,
+                    creationflags=_DETACHED_PROCESS,
+                    stdin=devnull, stderr=devnull, stdout=devnull)
+                _silence_resource_warning(popen)
+        except FileNotFoundError as exc:
+            raise Exception(
+                f'Failed to start {args[0]}: is it on your PATH: {exc}')
```
Comment (prashantmital) [inline pymongo/daemon.py:66]: How about `is it on your $PATH?\nOriginal exception: {exc}` ?

**Review by prashantmital — COMMENTED** [review]
Can you post the before and after error messages?

File: pymongo/daemon.py [hunk 2]
```diff
@@ -74,21 +78,28 @@ def _spawn_daemon(args):
...
-        # processes.
-        _popen_wait(temp_proc, _WAIT_TIMEOUT)
+        try:
+            temp_proc = subprocess.Popen(spawner_args, close_fds=True)
+            # Reap the intermediate child process to avoid creating zombie
+            # processes.
+            _popen_wait(temp_proc, _WAIT_TIMEOUT)
+        except FileNotFoundError as exc:
+            raise Exception(
+                f'Failed to start {args[0]}: is it on your PATH: {exc}')
```
Comment (ShaneHarvey) [inline pymongo/daemon.py:102]: We can revert the changes to this method. It's spawning this file itself which should always exist.

File: pymongo/daemon.py [hunk 3]
```diff
@@ -54,12 +54,16 @@ def _silence_resource_warning(popen):
...
-            _silence_resource_warning(popen)
+        try:
+            with open(os.devnull, 'r+b') as devnull:
+                popen = subprocess.Popen(
+                    args,
+                    creationflags=_DETACHED_PROCESS,
+                    stdin=devnull, stderr=devnull, stdout=devnull)
+                _silence_resource_warning(popen)
+        except FileNotFoundError as exc:
+            raise Exception(
```
Comment (ShaneHarvey) [inline pymongo/daemon.py:65]: Instead of Exception can we raise `PyMongoError`?

**Comment by guanlinzhou** [comment]
> Can you post the before and after error messages?

Before:
```
Traceback (most recent call last):
  File "/Users/william.zhou/mongo-python-driver/pymongo/daemon.py", line 122, in <module>
    _silence_resource_warning(_spawn(sys.argv[1:]))
  File "/Users/william.zhou/mongo-python-driver/pymongo/daemon.py", line 78, in _spawn
    return subprocess.Popen(
  File "/Users/william.zhou/opt/anaconda3/lib/python3.8/subprocess.py", line 854, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/Users/william.zhou/opt/anaconda3/lib/python3.8/subprocess.py", line 1702, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'mongocryptd'
```

After:
```
Traceback (most recent call last):
  File "/Users/william.zhou/mongo-python-driver/pymongo/daemon.py", line 129, in <module>
    _silence_resource_warning(_spawn(sys.argv[1:]))
  File "/Users/william.zhou/mongo-python-driver/pymongo/daemon.py", line 88, in _spawn
    raise Exception(
Exception: Failed to start path/path/path/path/path: is it on your PATH: [Errno 2] No such file or directory: 'mongocryptd'
Traceback (most recent call last):
  File "/Users/william.zhou/mongo-python-driver/pymongo/daemon.py", line 83, in _spawn
    return subprocess.Popen(
  File "/Users/william.zhou/opt/anaconda3/lib/python3.8/subprocess.py", line 854, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/Users/william.zhou/opt/anaconda3/lib/python3.8/subprocess.py", line 1702, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'mongocryptd'

File: pymongo/daemon.py (hunk 3 above)
Comment (ShaneHarvey) [reply pymongo/daemon.py:65]: I think you forgot about this one.

Comment (guanlinzhou) [reply pymongo/daemon.py]: yep just got to this

File: pymongo/daemon.py [hunk 4]
```diff
@@ -24,6 +24,8 @@
 import sys
 
 # The maximum amount of time to wait for the intermediate subprocess.
+from pymongo.errors import PyMongoError
```
Comment (ShaneHarvey) [inline pymongo/daemon.py:27]: Can you move the import up so that the comment above matches with `_WAIT_TIMEOUT`?

Comment (guanlinzhou) [reply pymongo/daemon.py:26]: fixed!

**Review by ShaneHarvey — APPROVED** [review]
LGTM!

**Review by prashantmital — APPROVED** [review]
